osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -47,7 +47,9 @@ public abstract partial class DrawableHitObject : PoolableDrawableWithLifetime<H
 
         /// <summary>
         /// The <see cref="HitObject"/> currently represented by this <see cref="DrawableHitObject"/>.
+        /// This may be null if the <see cref="DrawableHitObject"/> is in a pool and has not yet been applied a <see cref="HitObject"/>.
         /// </summary>
+        [CanBeNull]
```

Comment:
The nullability is fine and all, but if you're going to do this, you should also be updating all of the usages and there will be a _lot_ of them. There is eight in this file alone already:

![1717132522](https://github.com/ppy/osu/assets/20418176/77ad5f3d-bb7c-48e7-b36a-f05ec237c75b)

Please [run inspectcode locally](https://github.com/ppy/osu?tab=readme-ov-file#code-analysis) for a full list of these.